kythe/typescript/indexer.ts

```diff
@@ -30,6 +30,32 @@ const LANGUAGE = 'typescript';
  * code duplication.
  */
 export interface IndexerHost {
```

Comment:
This change seems to be expanding the mandate of IndexerHost.  Before, it was "stuff the indexer needs from its environment to do its job, with functions like emit()".  Now, it seems to encompass indexing work as well, representing the API between indexer and plugins.  I think it's useful to have that interface as well, but I don't think it all belongs here.  Another way of putting this is that I think it is now not possible for anyone to implement IndexerHost except for this particular file.

I think it'd be fine for the plugin to just receive a pointer to the indexer itself.

Reply:
Is there a reason that IndexerHost needs to be general enough for other files to implement?

The Plugin and the indexer need to be able to access the same canonical signature - that is the point of this change. The indexer seems to just be Visitor. Could a typed Indexer "pointer" just be passed to both the Visitor and the Plugin, with VName lookup methods/capabilities? Do you think that'd be more appropriate?

Reply:
I think introducing a separate indexer with VName lookup capabilities would make the ecosystem unnecessarily complicated.

Because the TS Indexer itself can be considered a plugin, another idea is to have a registry of indexers on the indexer host, similar to what Evan suggested. Plugins could then share arbitrary functionality. This may not be a nice API because it would likely have to be an array of `ts.Plugin`s that other plugins would have to search through, but it's worth considering.

Reply:
As discussed, IndexerHost is a misnomer. The add'l functionality is fine there.